### ISSUE TITLE
osu! window doesn't respond to refresh rate changes

### ISSUE THREAD

**Issue by nikunaku** [ppy/osu#27944]
### Type

Performance

### Bug description

As the title states, changing the refresh rate while osu! is running doesn't change the program's refresh rate. (I'm unsure if the same applies for VSync)

Eg. If I open osu! while my refresh rate is 60 Hz, and then I change my refresh rate to 144 Hz, osu! will remain at 60 Hz. This is fixed when I restart the program.

### Screenshots or videos

_(Using 2x frame limiter but it still applies)_

![osu-refresh-rate60 true-refresh-rate60](https://github.com/ppy/osu/assets/166950173/a9f9859a-e049-4796-a39c-e4dc08bcb4f3)
osu! and the monitor are both at 60 Hz

![osu-refresh-rate60 true-refresh-rate144](https://github.com/ppy/osu/assets/166950173/b4b66f6d-f5ab-4762-a0f5-ea41edb23cfe)
osu! is at 60 Hz even though the monitor got set to 144 Hz

![osu-refresh-rate144 true-refresh-rate144](https://github.com/ppy/osu/assets/166950173/85045e50-e91d-4613-87c9-a80c361cbdbd)
osu! and the monitor are now both at 144 Hz (osu! was restarted)

### Version

2024.412.1-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/15047870/compressed-logs.zip)

**Comment by honguyenminh** [comment]
I'm not sure whether this is a normal game behavior for other games, or even necessary at all. 

I tried a few games I have on my PC and none of them do this.

**Comment by peppy** [comment]
Yeah dunno, it's a super edge case. I'd expect it to maybe update when changing fullscreen mode at most.
